src/resources/Products/ProductInterfaces.ts

```diff
@@ -0,0 +1,209 @@
+import {ProductsSortCriteria, ProductsSortByType} from '../Enums';
+
+export interface AdvancedFiltersModel {
+    queryFilter: string;
+}
+
+export interface AdvancedParametersModel {
+    /**
+     * Used to retrieve debug information. Will have an impact on query performance.
+     */
+    debug?: boolean;
+    /**
+     * Used to retrieve all documents regardless of their security model.
+     */
+    viewAllContent?: boolean;
+}
+
+export interface FacetRequestOptions {
+    /**
+     * Whether to take into account the scores generated by the index when reordering facets.
+     */
+    enableIndexFacetOrdering?: boolean;
+    /**
+     * Whether facets should be returned in the same order in which they were requested.
+     */
+    freezeFacetOrder?: boolean;
+}
+
+export interface FacetRequestItem {
+    /**
+     * The values displayed by the facet in the search interface at the moment of the request.
+     */
+    currentValues?: any[];
+    /**
+     * A custom sort order for the facet values.
+     */
+    customOrder?: string[];
+    /**
+     * The unique identifier of the facet in the search interface.
+     */
+    facetId?: string;
+    /**
+     * The name of the field on which to base the facet request.
+     */
+    field?: string;
```

Comment:
Are all of those really optional? I would think you need at least the field?

Reply:
seems to be a mismatch between swagger and officiel model in our repo...